### ISSUE TITLE
Issue with Training Audio Only LTXV2 LoRA

### ISSUE THREAD

**Issue by thatrandomcatdude** [LongCatFan1/longcat-tuner#12]
Hi, I tried training ltx-v2 audio only lora with only .wav files, but after 2000 epochs, all I've got is motor noises :(

Any help would be appreciated.

* **Commit Hash**:  [c3f3cf0](https://github.com/AkaneTendo25/musubi-tuner/commit/c3f3cf044ef4f7633f1ebc6434e8c01e0d995f3f)
* **Training Dataset**: 10 audio .wav files with length of 5s each
* **Captioning**: All the .wav files captioned with only single sentence => `A woman talking in english with an indian accent.`
* **Commands run**:
```code
python ltx2_cache_latents.py \
--dataset_config dataset.toml \
--ltx2_checkpoint ltx-2-19b-dev-fp8.safetensors \
--device cuda \
--vae_dtype bf16 \
--ltx_mode audio \
--ltx2_audio_dir /mnt/wsl/PHYSICALDRIVE1p2/SD/musubi-tuner-forked/inputs/woman_audio \
--ltx2_audio_source audio_files

python ltx2_cache_text_encoder_outputs.py \
--dataset_config dataset.toml \
--ltx2_checkpoint ltx-2-19b-dev-fp8.safetensors \
--gemma_root /mnt/wsl/PHYSICALDRIVE1p2/SD/huggingface_cache/hub/models--google--gemma-3-12b-it-qat-q4_0-unquantized/snapshots/68f7ee4fbd59087436ada77ed2d62f373fdd4482 \
--gemma_load_in_8bit \
--device cuda \
--mixed_precision bf16 \
--ltx_mode audio \
--batch_size 1

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 ltx2_train_network.py \
--mixed_precision bf16 \
--dataset_config dataset.toml \
--fp8_base \
--gemma_load_in_8bit \
--gemma_root /mnt/wsl/PHYSICALDRIVE1p2/SD/huggingface_cache/hub/models--google--gemma-3-12b-it-qat-q4_0-unquantized/snapshots/68f7ee4fbd59087436ada77ed2d62f373fdd4482 \
--ltx2_checkpoint ltx-2-19b-dev-fp8.safetensors \
--ltx_mode audio \
--gradient_checkpointing \
--learning_rate 1e-4 \
--network_module networks.lora_ltx2 \
--network_dim 32 \
--network_alpha 32 \
--timestep_sampling shifted_logit_normal \
--sample_at_first \
--sample_every_n_epochs 5 \
--sample_with_offloading \
--sample_tiled_vae \
--sample_vae_tile_size 512 \
--sample_vae_tile_overlap 64 \
--sample_vae_temporal_tile_size 48 \
--sample_vae_temporal_tile_overlap 8 \
--output_dir output \
--output_name ltx2_lora \
--max_train_epochs 5000 \
--save_every_n_epochs 50 \
--optimizer_type adamw8bit \
--seed 42 \
--sdpa
```

* **dataset.toml**:
```toml
[general]
# resolution = 512
# caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = false

[[datasets]]
audio_directory = "/mnt/wsl/PHYSICALDRIVE1p2/SD/musubi-tuner-forked/inputs/woman_audio"
cache_directory = "/mnt/wsl/PHYSICALDRIVE1p2/SD/musubi-tuner-forked/inputs/woman_audio/cache" # recommended to set cache directory
caption_extension = ".txt"
num_repeats = 1
```

**Comment by thatrandomcatdude** [comment]
@m-i-i-m thank you for the suggestion. I'll try making these changes today and train it again. Will get back with the results

**Comment by LSI86** [comment]
> Thank for reporting, I will investigate this.

I have the same problem. I trained 2,000 samples, 25,000 steps, and 40 epochs. The result was absolutely zero. Is there a problem with assigning weights when loading it in comfyui? LTX2 reports as Flux when loading. Maybe if you don't have audio-video Lora but only audio, there may be a problem applying model weights?

**Comment by AkaneTendo25** [comment]
Sorry, I didn't have time to test extensively audio-only training yet. Perhaps there is some bug, I will indeed take a look at this later (but soon).

upd. About "LTX2 reports as Flux when loading" - not an error, is expected for LTX in ComfyUI (LTXV/LTXAV classes use ModelType.FLUX internally).

**Comment by m-i-i-m** [comment]
@LSI86 "Maybe if you don't have audio-video Lora but only audio, there may be a problem applying model weights?" This is not an issue, if I understand correctly. For loras that I generate that are "voice only", I blow away all the video layers and preserve only audio layers. This results in a super lightweight lora of 80-160 mb. There has been no issue loading and applying the lora in comfy.

**Comment by m-i-i-m** [comment]
@AkaneTendo25 , I tested an audio only dataset, training with wav audio files. This created a lora that generated audio that sounded like it was time stretched, slow. The same dataset created with an dummy video and setting fps, led to a lora that trained, just fine,  when only training for 1300 steps. When training with an audio only dataset, how would varying length audio clip be defined or bucketed?

**Comment by LSI86** [comment]
@m-i-i-m  May I ask for your functional workflow? I trained Audio Lora to improve the Czech language in the LTX2 model, but it has poor intonation and pronunciation, something like Ukrainian/Russian. That's why I'm surprised that such a large dataset and such extensive training has had no effect at all. Thanks for your help, I'm desperate. I rented a GPU CLOUD for training and trained for about 11 hours.

**Comment by m-i-i-m** [comment]
I treat all lora configuration as unique. I don't think it would help you. How large is your dataset and how are you captioning the files? I would presume that a generic language lora would be very hard to train. IMO, different from just training 1 specific person/character speaking Czech. You have 2000 audio files? Maybe provide your config, an audio from dataset, and and audio created from inference. I don't think what you are doing is going to be easy. I don't have experience with training languages.

**Comment by AkaneTendo25** [comment]
I found several more oversights that I had missed before, which were likely causing the slowdown and could prevent learning audio in some cases. I've committed fixes, but I can't test training right now since I'm stuck testing other modes, so I can't be 100% sure everything is reliably fixed yet.

**Comment by m-i-i-m** [comment]
@AkaneTendo25 I pulled your latest commit. The audio slowdown is still happening. It's not a feature that I need atm, though I wanted to test it out, since I hadn't updated the repo til recently. Let me know if you need some specific test to be run.

**Comment by m-i-i-m** [comment]
I do re-cache all the files on each test. I'm not sure if there are new settings that need to be set. I see you've added a lot more flags that are audio related. I'm not sure what they do yet. My test with audio with dummy video is 64x64 in dimension. When training, mode is set to audio, so in this case the video is ignored anyways if I understand correctly. For audio only training with "wav" dataset, my config is minimal as well. So not sure if I need to add any additional flags for things to work properly.

**Comment by AkaneTendo25** [comment]
You should not need extra flags for baseline audio-only training. Core `--ltx2_mode audio` training is expected to work with a minimal config. The newer audio flags are optional controls (mostly for bucketing/sampling behavior), not required settings.

I'll focus on fixing the audio-only path so it works reliably with default settings. Some of my design decisions may be wrong (for example, whether default bucketing should be pad or truncate), so I still need to validate this behavior.

**Comment by JonMonH** [comment]
Sorry I meant to send a heads up a little while ago about this. I've also found audio only training to still have the slowdown bug even after "coerce_decoded_audio_to_channels_first" was added to normalize the audio channels. It appears converting audio stream to a numpy array, while it might have the stereo data in arr.shape[1] it still reports the number of channels in arr.shape[0] as 1 while  the channels passed in through the function call has the correct number of channels. So none of the conditions are ever met and it just passes the array back. I just added a hack to make it work for now:

```
#arr reports the number of channels as 1 but the stream knows it's 2 channels
if arr.shape[0] < arr.shape[1] and arr.shape[0] != channels:
    return arr.reshape(-1, channels).T
```

I haven't tested any of the new stuff you've been working on, still on commit a65a207 in the Ltx2-dev branch. But I plan to test soon!

**Comment by AkaneTendo25** [comment]
@JonMonH You're right, my previous "fix" did not fix anything. Reusing your code snippet now.

**Comment by m-i-i-m** [comment]
@AkaneTendo25, I pulled your latest changes, and audio appears to be working with wav only dataset. I recached all files, trained for same samples and no slow down in voice.  I'll have to run some more test, but vram usage seems to be a bit higher than before. It's  using rough 5-6 gb more vram than before.

**Comment by AkaneTendo25** [comment]
@m-i-i-m Thanks for testing!
This increase in VRAM is strange. The commit 36d4b35 did not introduce any heavy VRAM-consuming operations. Are you sure you were always explicitly passing `--ltx2_mode audio` before?

**Comment by m-i-i-m** [comment]
Yes, the "--ltx2_mode audio" was set in my batch script, only difference being, how I cached the audio  wav/ audio with dummyvid.
Maybe I'm just loosing my mind, lol. Audio only training should use less memory right? I should probably save like a memory usage log with my training tests. --- Yeah sorry for the noise, I am doing so many different test, from image only training t2v, ic lora testing, etc. My brain is just mixing things up.
